### PR TITLE
Feature/e2e performance tests

### PR DESCRIPTION
**Pull request by jduffy3**

### CODE REVIEW

**Review by sbrudz — CHANGES_REQUESTED** [review]

File: src/components/JustNotSorry.js [hunk 1]
```diff
@@ -49,17 +49,25 @@ const JustNotSorry = ({ onEvents, phrases }) => {
...
-        key={index}
-        textArea={parentRect}
-        range={warning.rangeToHighlight}
-        message={warning.message}
-        number={index}
-      />
-    ));
-    return ReactDOM.createPortal(warningComponents, email.current.offsetParent);
+    const currentEmail = email.current;
+    const parentRect = currentEmail.offsetParent.getBoundingClientRect();
```
[the remark this answers is not in the document]
Comment (jduffy3) [reply src/components/JustNotSorry.js:53]: Strange I'm not getting the above error and I tested via `npm run start:chrome` and loading the unpacked lib into my own browser... 😕

[the remark this answers is not in the document]
Comment (sbrudz) [reply src/components/JustNotSorry.js:53]: That is strange.  I used the same command as you.

I first saw the error when I opened an email in my drafts folder after composing the first email (step 1).  Do you see the error if you try that as step 2?

Also, my `chrome://version` returns:

Google Chrome | 110.0.5481.100 (Official Build) (arm64)
-- | --
Revision | 4be7a36f7cb943af6118e449bbab494b43dcaddd-refs/branch-heads/5481_77@{#14}
OS | macOS Version 12.6.3 (Build 21G419)
JavaScript | V8 11.0.226.16
User Agent | Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36

What are you running?

[the remark this answers is not in the document]
Comment (jduffy3) [reply src/components/JustNotSorry.js:53]: Google Chrome | 110.0.5481.177 (Official Build) (arm64)
-- | --
Revision | f34f7ab2d4ca4ad498ef42aeba4f4eb2c1392d63-refs/branch-heads/5481@{#1239}
OS | macOS Version 13.0.1 (Build 22A400)
JavaScript | V8 11.0.226.16
User Agent | Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36


😕

[the remark this answers is not in the document]
Comment (sbrudz) [reply src/components/JustNotSorry.js:53]: Yes, this is from Gmail.  I wouldn't think that the version of MacOS would make a difference, but maybe...?

[the remark this answers is not in the document]
Comment (jduffy3) [reply src/components/JustNotSorry.js:53]: Not sure :( I pushed up a change that checks that its not null or undefined on the ref and added some setup scripts around modifying the manifest 🙏

**Comment by jduffy3** [comment]
Thanks @sbrudz  the video was super helpful. So the offset parent is null when the message gets minimized when composing a new email. My window was wide enough to keep the two messages open, that's why it wasn't failing!

**Review by sbrudz — APPROVED** [review]
I didn't think about window size being a factor.  The fix looks good.  No more errors.  Thank you!  :shipit:

**Pull request merged**